src/Data/List/Relation/Unary/Any/Properties.agda

```diff
@@ -255,24 +262,41 @@ Any-×⁻ pq with Product.map₂ (Product.map₂ find) (find pq)
 
     (p , q) ∎
 
-  to∘from : ∀ pq → Any-×⁺ {xs = xs} (Any-×⁻ pq) ≡ pq
-  to∘from pq with find pq
-      | (λ (f : (proj₁ (find pq) ≡_) ⋐ _) → map∘find pq {f})
-  ... | (x , x∈xs , pq′) | lem₁
-    with find pq′
-      | (λ (f : (proj₁ (find pq′) ≡_) ⋐ _) → map∘find pq′ {f})
-  ... | (y , y∈ys , p , q) | lem₂
-    rewrite sym $ map-∘ {R = λ x → Any (λ y → P x × Q y) ys}
-                        (λ p → Any.map (λ q → p , q) (lose y∈ys q))
-                        (λ y → subst P y p)
-                        x∈xs
-            = lem₁ _ helper
-    where
-    helper : Any.map (λ q → p , q) (lose y∈ys q) ≡ pq′
-    helper rewrite sym $ map-∘ (λ q → p , q)
-                               (λ y → subst Q y q)
-                               y∈ys
-           = lem₂ _ refl
+  to∘from : ∀ pq → Any-×⁺ (Any-×⁻ pq) ≡ pq
+  to∘from pq =
+    let x , x∈xs , pq′ = find pq
+
+        lem₁ : {f : (x ≡_) ⋐ λ x → Any (λ y → (P x) × (Q y)) ys} →
+               (f-refl : f refl ≡ pq′) → Any.map f _ ≡ pq
+        lem₁ {f} = map∘find pq {f = f}
```

Comment:
Looking at this again: if the proof is `map∘find` why doesn't the statement seem to have a `find` in it? [My guess: the statement has an unfolding of the definition of `find` in it.]  So perhaps the lemma statement would be clearer without doing as much expansion?

Reply:
This whole proof used to be incomprehensible because of the *very* tricky use of inlined-`with` to introduce lemmas whose types aren't well-formed, but whose definitions are (!?!?!?), and this very verbose version is the best I could do to reconstruct that. The use of `_` hides ...

Reply:
Thanks for continuing to nudge me towards the final version with each of `lem₁` and `lem₂` inlined, I hope it's much clearer now.

The `map∘find` arises not out of excessive expansion, but because `lose` is a disguised version of `map`on the outcome of a `find` (so *its* definition might one day usefully be inlined, but I really don't want that deprecation headache this time around...) and I think the goals are a bit confusing because of a ``map∘find∘map` composiiton which might yield to a different analysis... but again, for another time!

Reply:
Resolved to your satisfaction @JacquesCarette ?

Reply:
This is good enough. Further 'improvements' should be done by me.